libexec/cli/action_argparser.sh

```diff
@@ -116,22 +116,24 @@ while true; do
         ;;
         -n|--nv)
             shift
-            for i in `ldconfig -p | grep -E "/libnv|/libcuda|/libEGL|/libGL|/libnvcu|/libvdpau|/libOpenCL|/libOpenGL"`; do
-                if [ -f "$i" ]; then
-                    message 2 "Found NV library: $i\n"
-                    if [ -z "${SINGULARITY_CONTAINLIBS:-}" ]; then
-                        SINGULARITY_CONTAINLIBS="$i"
-                    else
-                        SINGULARITY_CONTAINLIBS="$SINGULARITY_CONTAINLIBS,$i"
+            for lib in $(cat $SINGULARITY_sysconfdir"/singularity/nvliblist.conf" | grep -v "^#"); do
+                for i in $(ldconfig -p | grep "${lib}"); do
```

Comment:
I'm not sure if I follow what your thinking.  The first loop iterates over the library patterns in nvliblist.conf while the second loop iterates over the matches in the ld.so.cache.  Sorry if I'm missing something easy, but I don't see how it could be done with a single loop.